### PR TITLE
Tools: autotest.py: look at EKF flags to determine armability

### PR DESCRIPTION
**Pull request by peterbarker**
This aids in increasing speedup as we no longer rely on a `statustext` coming in after we're waiting for armability - at higher speedups that can come in before we are listening for it.

If we ever publish a message actually indicating armability we should switch to using that message...

### CODE REVIEW

**Comment by khancyr** [comment]
I like it , pymavlink is also doing it, if it solve some part of the timing issue I am in favor of it.
But I also agree with Fransisco comment

**Comment by peterbarker** [comment]
@OXINARF *thought* this felt awfully familiar...

**Comment by peterbarker** [comment]
@OXINARF @khancyr Another (squashable) patch pushed up.  No timeout by default, and we raise an exception on timeout, none of this true-false thing in an assertively named function call...

File: Tools/autotest/common.py [hunk 1]
```diff
@@ -10,6 +10,28 @@
...
+def wait_ekf_happy(mav, timeout=30):
+    """Wait for EKF to be happy"""
+
+    tstart = get_sim_time(mav)
+    required_value = 831
+    print("Waiting for EKF value %u" % (required_value))
+    while timeout is None or get_sim_time(mav) < tstart + timeout:
+        m = mav.recv_match(type='EKF_STATUS_REPORT', blocking=True)
+        current = m.flags
+        print("Wait EKF.flags: required:%u current:%u" % (required_value, current))
```
Comment (OXINARF) [inline Tools/autotest/common.py]: We should either don't print this message or at least rate limit it. Just in our short Travis autotest job it got 337 instances of this message.

**Comment by peterbarker** [comment]
On Tue, 12 Sep 2017, Francisco Ferreira wrote:

> We should either don't print this message or at least rate limit it. Just in
> our short Travis autotest job it got 337 instances of this message.

They're good for you.  Just like all of the other "waiting for" messages. 
I did start a generic "self.progress(...)" thing, maybe I should look at 
pushing that forward.

**Comment by peterbarker** [comment]
@OXINARF rate-limitting added.  Squash away :-)

**Comment by OXINARF** [comment]
Much better, only 3 messages now! Thanks.

Squashed and merged, thank you!

**Pull request closed without merging**